patch/patches/content_main_654986.patch

```diff
@@ -39,7 +39,46 @@ index 96c28a7ce3183..3d60ab170e9a5 100644
  
    // A flag to indicate whether Main() has been called before. On Android, we
    // may re-run Main() without restarting the browser process. This flag
-@@ -274,14 +268,6 @@ RunContentProcess(ContentMainParams params,
+@@ -252,21 +246,23 @@ RunContentProcess(ContentMainParams params,
+ // On Android setlocale() is not supported, and we don't override the signal
+ // handlers so we can get a stack trace when crashing.
+ #if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_ANDROID)
+-    // Set C library locale to make sure CommandLine can parse
+-    // argument values in the correct encoding and to make sure
+-    // generated file names (think downloads) are in the file system's
+-    // encoding.
+-    setlocale(LC_ALL, "");
+-    // For numbers we never want the C library's locale sensitive
+-    // conversion from number to string because the only thing it
+-    // changes is the decimal separator which is not good enough for
+-    // the UI and can be harmful elsewhere. User interface number
+-    // conversions need to go through ICU. Other conversions need to
+-    // be locale insensitive so we force the number locale back to the
+-    // default, "C", locale.
+-    setlocale(LC_NUMERIC, "C");
+-
+-    SetupSignalHandlers();
++    if (!params.disable_signal_handlers) {
++        // Set C library locale to make sure CommandLine can parse
++        // argument values in the correct encoding and to make sure
++        // generated file names (think downloads) are in the file system's
++        // encoding.
++        setlocale(LC_ALL, "");
```

Comment:
Is it necessary to also remove these `setlocale` calls?

Reply:
Most likely not. Will move them out and move the signal comments underneath